eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementOutputLibrary.cs

```diff
@@ -149,11 +149,11 @@ static Dictionary<ResourceScope, ResourcesAndNonResourceMethodsInScope> BuildRes
                     [ResourceScope.ResourceGroup] = new([], [], []),
                     [ResourceScope.Subscription] = new([], [], []),
                     [ResourceScope.Tenant] = new([], [], []),
-                    [ResourceScope.ManagementGroup] = new([], [], []),
+                    [ResourceScope.ManagementGroup] = new([], [], [])
                 };
                 foreach (var (metadata, resourceClient) in resourceDict)
                 {
-                    if (metadata.ParentResourceId is null)
+                    if (metadata.ParentResourceId is null && metadata.ResourceScope != ResourceScope.Extension)
```

Comment:
since here we are categorizing the resources by their types, I think we do not have to exclude extension resources.
Keeps tracking them could help.
Also, we could have non-resource "extension" operations.

So I think we should have an extra type here to be `Extension`, and when we consume it above, we should put them inside the MockableArmClient class.